File: Programmers/2/42578. 의상/의상.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(String[][] clothes) {
        /* 
        int answer = 1;
        
        Map<String, Integer> map = new HashMap<>();
        
        for (String[] cloth : clothes) {
            String category = cloth[1];
            
            map.put(category, map.getOrDefault(category, 0) + 1);
        }
        
        for (int count : map.values()) {
            answer *= (count + 1);
        }
        
        return answer - 1;
        */
        
        return Arrays.stream(clothes)
            .collect(Collectors.groupingBy(cloth -> cloth[1], Collectors.counting()))
            .values()
            .stream()
            .map(count -> count + 1)
            .reduce(1L, (a, b) -> a * b)
            .intValue() - 1;
        }
}

File: Programmers/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int result = 1;
    unordered_map<string, int> mp;

    for (auto &c : clothes) {
        mp[c[1]]++;
    }

    for (auto &p : mp) {
        result *= (p.second + 1);
    }

    return result - 1;
}